runtime/vam/expr/genarithfuncs.go

```diff
@@ -0,0 +1,133 @@
+//go:build ignore
+
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"go/format"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/brimdata/zed/vector"
+)
+
+var opToAlpha = map[string]string{
+	"+": "Add",
+	"-": "Sub",
+	"*": "Mul",
+	"/": "Div",
+	"%": "Mod",
+}
+
+func main() {
+	var buf bytes.Buffer
+	fmt.Fprintln(&buf, "// Code generated by genarithfuncs.go. DO NOT EDIT.")
+	fmt.Fprintln(&buf)
+	fmt.Fprintln(&buf, "package expr")
+	fmt.Fprintln(&buf, "import (")
+	fmt.Fprintln(&buf, `"github.com/brimdata/zed/vector"`)
+	fmt.Fprintln(&buf, `"github.com/brimdata/zed"`)
+	fmt.Fprintln(&buf, ")")
+
+	var ents strings.Builder
+	for _, op := range []string{"+", "-", "*", "/", "%"} {
+		for _, typ := range []string{"Int", "Uint", "Float", "String"} {
+			if typ == "Float" && op == "%" ||
+				typ == "String" && op != "+" {
+				continue
+			}
+			for lform := vector.Form(0); lform < 3; lform++ {
+				for rform := vector.Form(0); rform < 3; rform++ {
+					name := "arith" + opToAlpha[op] + typ + lform.String() + rform.String()
```

Comment:
Your naming convention differs from your compare pr in that the generated functions are seperated by underscores (e.g., `cmp_EQ_Int_Flat_Flat`). I think you should pick one for consistency.

Reply:
I'm planning to remove the underscores from gencomparefuncs.go in a separate PR.